tce/src/applibs/LLVMBackend/TDGen.cc

```diff
@@ -3474,7 +3474,7 @@ TDGen::writeInstrInfo(std::ostream& os) {
     writeControlFlowInstrDefs(os);
 
     // Hardware loop instructions
-    if (mach_.controlUnit()->hasOperation("hwloop")) writeHWLoopDef(os);
+    writeHWLoopDef(os);
```

Comment:
Is this on purpose? It now always writes the HWLoopDef even if the ADF doesn't have it?

Reply:
Inside `writeHWLoopDef` we have a condition to check "hwloop" instruction in the adf.
However, we have to add `LJUMP` instruction to TDGen irrespective of `hwloop` instruction. 

We are doing custom instr-selection for LJUMP. So, we need to hard-code its opcode (TCE::LJUMP) in the TCEISelector.
That is why we moved this "hwloop" check inside writeHWLoopDef().